### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by ElTh0r0**
Hello,
I'm not sure, if you just want to preserve the code from the Qt wiki or if you are interested in some changes/corrections. Feel free to merge my changes.

### CODE REVIEW

File: dialog.cpp [hunk 1]
```diff
@@ -22,8 +23,34 @@ Dialog::~Dialog()
 
 void Dialog::checkSpelling()
 {
-    QString dictPath = "/usr/share/hunspell/hu_HU";
-    QString userDict= "/usr/share/hunspell/hu_HU.aff";
+    QString language = "en_US";
+
+    QString dictPath = "/usr/share/hunspell";
+    if (!QDir(dictPath).exists()) {
```
Comment (martonmiklos) [inline dialog.cpp]: Should not we wrap these platform dependant sections between #ifdef Q_OS_LINUX/WIN32 defines?

File: dialog.cpp [hunk 2]
```diff
@@ -22,8 +23,34 @@ Dialog::~Dialog()
...
+        dictPath = QStringLiteral("/usr/local/share/hunspell/");
+        // Otherwise look for MySpell dictionary
+      } else if (QDir(QStringLiteral("/usr/share/myspell/dicts")).exists()) {
+        dictPath = QStringLiteral("/usr/share/myspell/dicts/");
+      } else if (QDir(
+                   QStringLiteral("/usr/local/share/myspell/dicts")).exists()) {
+        dictPath = QStringLiteral("/usr/local/share/myspell/dicts/");
+      } else {
+        // Fallback and for Windows: Use app dir
+        dictPath = qApp->applicationDirPath() + "/dicts/";
```
Comment (martonmiklos) [inline dialog.cpp]: On Windows the directory separator is the opposite slash (\)

File: dialog.cpp [hunk 3]
```diff
@@ -22,8 +23,34 @@ Dialog::~Dialog()
...
+      }
+    }
+    if (!dictPath.endsWith('/')) {
```
Comment (martonmiklos) [inline dialog.cpp]: I think we should rather use QDir::separator here.

File: dialog.cpp [hunk 4]
```diff
@@ -87,6 +114,9 @@ void Dialog::checkSpelling()
                 case SpellCheckDialog::ReplaceOnce:
                     cursor.insertText(checkDialog->replacement());
                     break;
+                case SpellCheckDialog::ReplaceAll:
+                    this->replaceAll(cursor.position(), word, checkDialog->replacement());
```
Comment (martonmiklos) [inline dialog.cpp]: I think we should remove the this-> part (just to be consistent with the rest of the code.)

File: dialog.cpp [hunk 5]
```diff
@@ -103,3 +133,30 @@ void Dialog::checkSpelling()
                 tr("Finished"),
                 tr("The spell check has finished."));
 }
+
+
+void Dialog::replaceAll(int nPos, QString sOld, QString sNew) {
```
Comment (martonmiklos) [inline dialog.cpp]: use const QString & for argument types

File: dialog.cpp [hunk 6]
```diff
@@ -103,3 +133,30 @@ void Dialog::checkSpelling()
...
+    QTextCursor cursor(ui->textEdit->document());
+    cursor.setPosition(nPos-sOld.length(), QTextCursor::MoveAnchor);
+
+    while(!cursor.atEnd()) {
```
Comment (martonmiklos) [inline dialog.cpp]: Please add a space between the while and the '('.
I know the code has some occurences where no space was used, however the official Qt coding style employs that.

File: dialog.cpp [hunk 7]
```diff
@@ -103,3 +133,30 @@ void Dialog::checkSpelling()
...
+        QCoreApplication::processEvents();
+        cursor.movePosition(QTextCursor::EndOfWord, QTextCursor::KeepAnchor, 1);
+        QString word = cursor.selectedText();
+
+        // Workaround for better recognition of words
+        // punctuation etc. does not belong to words
+        while(!word.isEmpty() && !word.at(0).isLetter() && cursor.anchor() < cursor.position()) {
```
Comment (martonmiklos) [inline dialog.cpp]: Space between the while and the (

File: dialog.cpp [hunk 8]
```diff
@@ -103,3 +133,30 @@ void Dialog::checkSpelling()
...
+            int cursorPos = cursor.position();
+            cursor.setPosition(cursor.anchor() + 1, QTextCursor::MoveAnchor);
+            cursor.setPosition(cursorPos, QTextCursor::KeepAnchor);
+            word = cursor.selectedText();
+        }
+
+        if(word == sOld) {
```
Comment (martonmiklos) [inline dialog.cpp]: Add a space between the if and the (

**Comment by martonmiklos** [comment]
Hello @ElTh0r0 

Many thanks for your fixes! 

My basic motivation with this repo was the following:
When I wanted to work with Qt + hunspell, so I looked for examples. Getting the example was hard so I created this repo to make the life of the future newcomers easier. Your changes aligns with this effort, so I reviewed it and left some feedback.

File: dialog.cpp (hunk 2 above)
Comment (ElTh0r0) [reply dialog.cpp]: "/" works universal in Qt. From the [QDir::separator documentation](https://doc.qt.io/qt-5/qdir.html#separator): _You do not need to use this function to build file paths. If you always use "/", Qt will translate your paths to conform to the underlying operating system._

File: dialog.cpp (hunk 3 above)
Comment (ElTh0r0) [reply dialog.cpp]: "/" works universal in Qt. From the [QDir::separator documentation](https://doc.qt.io/qt-5/qdir.html#separator): _You do not need to use this function to build file paths. If you always use "/", Qt will translate your paths to conform to the underlying operating system._

File: dialog.cpp (hunk 1 above)
Comment (ElTh0r0) [reply dialog.cpp]: #ifdef would be possible to make it more clear which path belongs to which operating system, but according to my knowledge the application is not running into issues when it is executed on Windows (it just would not find the folder).

**Comment by ElTh0r0** [comment]
Many thanks for your constructive feedback @martonmiklos ! Initially I didn't want to make these big changes (at least they seem to be big, but most of the linter changes are related to indents), but I think it makes sense to create pretty code by using cpplint + clazy and replace deprecated Qt features.
Please make a short review again!

File: dialog.cpp (hunk 2 above)
Comment (martonmiklos) [reply dialog.cpp]: > "/" works universal in Qt. From the [QDir::separator documentation](https://doc.qt.io/qt-5/qdir.html#separator): _You do not need to use this function to build file paths. If you always use "/", Qt will translate your paths to conform to the underlying operating system._

Ah thanks I missed this point!

File: SpellChecker.pro [hunk 9]
```diff
@@ -1,4 +1,7 @@
 QT      += core gui widgets
+greaterThan(QT_MAJOR_VERSION, 5) {
+  QT    += core5compat
```
Comment (martonmiklos) [inline SpellChecker.pro:3]: Which used API is not supported in Qt6? i.e. if you remove this section what error do you get? (I have to admit I do not have Qt6 installed yet :))

**Comment by ElTh0r0** [comment]
QTextCodec is not existing anymore in Qt6. Replacement should be QStringConverter ([see Qt blog post](https://www.qt.io/blog/porting-from-qt-5-to-qt-6-using-qt5compat-library)), but for simple compatibility for Qt 5 **and** 6 with the same source code, the most easiest solution is to add "core5compat".
If I find some time, I'll check if there is a "pure" Qt 6 solution with above mentioned new class as well, but I think for the meantime using core5compat is fine.

File: SpellChecker.pro (hunk 9 above)
Comment (ElTh0r0) [reply SpellChecker.pro:3]: QTextCodec is not existing anymore in Qt6. Replacement should be QStringConverter ([see Qt blog post](https://www.qt.io/blog/porting-from-qt-5-to-qt-6-using-qt5compat-library)), but for simple compatibility for Qt 5 **and** 6 with the same source code, the most easiest solution is to add "core5compat".
If I find some time, I'll check if there is a "pure" Qt 6 solution with above mentioned new class as well, but I think for the meantime using core5compat is fine.

Comment (martonmiklos) [reply SpellChecker.pro:3]: > If I find some time, I'll check if there is a "pure" Qt 6 solution with above mentioned new class as well, but I think for the meantime using core5compat is fine.
Leave as it is now, I was just curious. Thanks for the contribution!

**Pull request merged**